tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4679,6 +4693,8 @@ TEST_F(ValidationTest, PSVStringTableReorder) {
        "')",
        "error: DXIL container mismatch for 'EntryFunctionName' between 'PSV0' "
        "part:('ain') and DXIL module:('main')",
+       "error: In 'StringTable', 'A' is not used",
```

Comment:
Looking at your verify function at `lib/DxilValidation/DxilContainerValidation.cpp:117`, there should be an error here for `B` not being used, right?

Reply:
Presumably because `tools/clang/unittests/HLSL/ValidationTest.cpp::4625` has the start of the string table as 'B', 0.

Reply:
Hey Xiang, can you respond to this? I think an error should be emitted that is not being emitted.

Reply:
Sorry, I thought you've explained it.

Here's more detail.
The OrigStringTable is {0,   'A', 0,   'B', 0, 'm', 'a', 'i', 'n', 0,   0, 0};
'', 'A', 'B' and 'main' are used which map to index 0, 1, 3, 5

The UpdatedStringTable is {'B', 0,   'A', 0, 'm', 'a',  'i', 'n', 0,   0, 0,   0};
the index 0, 1, 3, 5 now map to 'B', '', '', 'ain'.
So 'B' is used like you said above.

Reply:
Ah I think I get it. The OffsetToUseCountMap is not being reset, but the Table is.